app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/template.rb

```diff
@@ -56,12 +56,13 @@ def self.raw_import_image(ext_management_system, options = {})
     wrkfl_timeout = options['timeout'].to_i.hours
 
     location, node_auth, _ = node_creds(options['src_provider_id'])
-    guid, apikey, region, endpoint, access_key, secret_key = cos_creds(options['obj_storage_id'])
+    guid, apikey, region, endpoint, _, _ = cos_creds(options['obj_storage_id'])
     bucket = bucket_name(options['bucket_id'])
     diskType = CloudVolumeType.find(options['disk_type_id']).name
 
     cos_ans_creds = {:resource_instance_id => guid, :apikey => apikey, :bucket_name => bucket, :url_endpoint => endpoint}
-    cos_pvs_creds = {:region => region, :bucketName => bucket, :accessKey => access_key, :secretKey => secret_key}
+    cos_bearer_id = Authentication.find_by(:resource_id => options['obj_storage_id'], :authtype => 'bearer').id
```

Comment:
Similar comment here...this feels like we need more scoping on the query itself, even though this only returns an id.

Reply:
`resource_id` isn't sufficient to query a polymorphic association, you could hit on a different resource_type with the resource_id you're looking for.

And yeah it is possible the auth record could be deleted after this was queued and before it is processed so it could fail to be found.

Reply:
Ok, I thought the IDs were unique regardless of the polymorphic type. I see now. 

The second part is a valid concern, only don't know of methods to check for existence and and get the record atomically. I guess doing "find(...)" would just throw an exception if the record is not found, which is close to what we need. It is "find!" that would return nil in such case from what I know.